dbms/src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -789,6 +790,27 @@ void MergeTreeData::loadDataParts(bool skip_sanity_checks)
 
     auto disks = storage_policy->getDisks();
 
+    if (getStoragePolicy()->getName() != "default")
```

Comment:
Hm, why this sanity is not performed for `default` policy? (since it can be overwritten to use multiple volumes/disks) Am I missing something?

And speaking about this, should this check be done under `skip_sanity_checks`/`force_restore_data`?

Reply:
I have asked @excitoon and he answered "it needs a fix".

Reply:
Fixed/fixing in #8440